syntax = "proto3";

package eu.surething_project.core;

option java_multiple_files = true;
option java_package = "eu.surething_project.core.grpc";
option java_outer_classname = "LocEndorseProto";

import "google/protobuf/any.proto";
import "loc_time.proto";
import "signature.proto";

// Location Endorsement generated by the witness to testify a prover's location claim(s)
message LocationEndorsement{
	// witness unique id
	string witnessId = 1;
	// the original location claim of the prover
	string claimId = 2;
	// time of location endorsement - in one of alternative time schemas
	Time time = 3;
	// type of the evidence
	string evidenceType = 4;
	// endorsement evidence
	google.protobuf.Any evidence = 5;
}

message SignedLocationEndorsement{
	LocationEndorsement endorsement = 1;
	Signature witnessSignature = 2;
}
